apps/omg/test/omg/fees_test.exs

```diff
@@ -54,30 +54,35 @@ defmodule OMG.FeesTest do
     @payment_tx_type => @payment_fees
   }
 
-  describe "covered?/2" do
+  describe "check_if_covered/2" do
     test "does not check the fees when :no_fees_required is passed" do
-      assert Fees.covered?(%{@eth => 0}, :no_fees_required)
+      assert Fees.check_if_covered(%{@eth => 0}, :no_fees_required) == :ok
     end
 
-    test "returns true when fees are covered by another currency" do
-      assert Fees.covered?(%{@not_eth_1 => 5}, @payment_fees)
+    test "returns :ok when fees are exactly covered by one currency" do
+      assert Fees.check_if_covered(%{@not_eth_1 => 3, @eth => 0}, @payment_fees) == :ok
     end
 
-    test "returns true when multiple implicit fees are given and fee is covered by eth" do
-      assert Fees.covered?(%{@eth => 2, @not_eth_1 => 2}, @payment_fees)
+    test "returns :multiple_potential_currency_fees when multiple implicit fees are given" do
+      assert Fees.check_if_covered(%{@eth => 2, @not_eth_1 => 2}, @payment_fees) ==
+               {:error, :multiple_potential_currency_fees}
     end
 
-    test "returns true when multiple implicit fees are given and fee is covered by another currency" do
-      assert Fees.covered?(%{@eth => 0.5, @not_eth_1 => 4}, @payment_fees)
+    test "returns fees_not_covered when the implicit fees currency does not match any of the supported fee currencies" do
+      other_currency = <<2::160>>
+      assert Fees.check_if_covered(%{other_currency => 100}, @payment_fees) == {:error, :fees_not_covered}
     end
 
-    test "returns false when the implicit fees currency does not match any of the supported fee currencies" do
-      other_currency = <<2::160>>
-      refute Fees.covered?(%{other_currency => 100}, @payment_fees)
+    test "returns fees_not_covered when fees do not cover the fee price" do
+      assert Fees.check_if_covered(%{@not_eth_1 => 1}, @payment_fees) == {:error, :fees_not_covered}
+    end
+
+    test "returns overpaying_fees when fees cover more than the fee price" do
+      assert Fees.check_if_covered(%{@not_eth_1 => 4}, @payment_fees) == {:error, :overpaying_fees}
     end
```

Comment:
the new behavior added to the module is that `0` fees "don't count" and don't allow transactions through, I think this isn't tested.

BTW, are we testing for existence `: 0` entries in the fee specs early on, and rejecting such fee specs or we're ok with such entries?

Reply:
We were ok until now, will update the parser in this PR to reject `0` fee amounts